File: src/main/java/io/vproxy/vfx/animation/AnimationStateTransferFinishCallback.java
package io.vproxy.vfx.animation;

import io.vproxy.vfx.util.algebradata.AlgebraData;

public interface AnimationStateTransferFinishCallback<T extends AlgebraData<T>> {
    void animationStateTransferFinish(AnimationNode<T> from, AnimationNode<T> to);
}

File: src-ts/src/animation/AnimationStateTransferFinishCallback.ts
import type { AlgebraData } from '../util/algebradata/AlgebraData.js';
import type { AnimationNode } from './AnimationNode.js';

export type AnimationStateTransferFinishCallback<T extends AlgebraData<T>> =
  (from: AnimationNode<T>, to: AnimationNode<T>) => void;
